UMN-AutoPackager/public/Set-UMNGlobalConfig.ps1

```diff
@@ -0,0 +1,153 @@
+        }
+    end {
+        Write-Verbose -Message "Ending $($myinvocation.mycommand)"
+    }
+}#Get-Config
+function Set-UMNGlobalConfig {
+    <#
+    .SYNOPSIS
+        Sets a value(s) for the global configuration JSON file for the auto packager.
+    .DESCRIPTION
+        This command sets in a json file a value for one of various global configurations used as part of the AutoPackager. It requires a the full path and name of the JSON file, the key to set, and the value to set it to.
+    .EXAMPLE
+        Set-UMNGlobalConfig -Path .\config.json -Key CompanyName  -Value "University of Minnesota"
+        Sets the value of CompanyName to University of Minnesota for the config.json file
+    .EXAMPLE
+        Set-UMNGlobalConfig -Path .\config.json -SiteServer "my.config.site"  -SiteCode "COM" -DownloadLocationPath "C:\\Temp" -ApplicatoinContentPath "\\appstorage.somewhere\"
+        Sets the values for a ConfigMgr site to be added to the config.json file
+    .PARAMETER Path
+        The full path and file name of the JSON file to be updated
+    .PARAMETER Key
+        The key that you want to update in the Global Configuation JSON file
+    .PARAMETER Value
+        The value you want to set the key to
+    .PARAMETER SiteServer
+        The ConfigMgr site server address
+    .PARAMETER SiteCode
+        The site code for this instance of ConfigMgr
+    .PARAMETER DownloadLocationPath
+        The full path where downloaded data will be temporarily stored
+    .PARAMETER ApplicationContentPath
+        The full path for where ConfigMgr applcation content will be stored"
+    #>
+    [CmdletBinding(DefaultParameterSetName="Main")]
+    param (
+        [Parameter(ParameterSetName="Main",Mandatory=$True, HelpMessage = "The full path and file name of the JSON file to be updated")]
+        [Parameter(ParameterSetName="ConfigMgr")]
+        [ValidateNotNullOrEmpty()]
+        [string]$Path,
+        [Parameter(ParameterSetName="Main",Mandatory=$True, HelpMessage = "The key that you want to update in the Global Configuation JSON file")]
+        [ValidateNotNullOrEmpty()]
+        [string]$Key,
+        [Parameter(ParameterSetName="Main",Mandatory=$True, HelpMessage = "The value you want to set the key to")]
+        [ValidateNotNullOrEmpty()]
+        [string]$Value,
+        [Parameter(ParameterSetName="ConfigMgr",Mandatory=$True, HelpMessage = "The ConfigMgr site server address")]
+        [string]$SiteServer,
+        [Parameter(ParameterSetName="ConfigMgr",Mandatory=$True, HelpMessage = "The site code for this instance of ConfigMgr")]
+        [string]$SiteCode,
+        [Parameter(ParameterSetName="ConfigMgr",Mandatory=$True, HelpMessage = "The full path where downloaded data will be temporarily stored")]
+        [string]$DownloadLocationPath,
+        [Parameter(ParameterSetName="ConfigMgr",Mandatory=$True, HelpMessage = "The full path for where ConfigMgr applcation content will be stored")]
+        [string]$ApplicationContentPath
```

Comment:
For future revisions, we may want to add functionality to ensure paths have / and \ escaped either as validated input or when writing values to the json file.

Reply:
Resolving for now. If it needs to be an Enhancement later, we can open an issue for it then. This doesn't need to hold up approval. Resolving.